src/protocol/context/semi_honest.rs

```diff
@@ -32,12 +32,15 @@ impl<'a, F: Field> SemiHonestContext<'a, F> {
     }
 
     #[must_use]
-    pub fn upgrade_to_malicious(
+    pub fn upgrade<S: Substep + ?Sized>(
         self,
-        accumulator: SecurityValidatorAccumulator<F>,
+        protocol_step: &S,
```

Comment:
I don't love this name. It's very generic and it doesn't tell me how it'll be used.

Reply:
Perhaps something like `mirror_step` or `duplicated_circuit_step`